SofaScoreAcademy/SofaScoreAcademy/Controllers/LeagueMatchesViewController.swift

```diff
@@ -13,37 +13,73 @@ class LeagueMatchesViewController: UIViewController {
     
     var sportSlug: String? {
         didSet {
-            updateForSelectedSport()
+            fetchEvents()
         }
     }
-
-    private var exampleEvents: [Event] = []
+    var date: Date? {
+        didSet {
+            fetchEvents()
+        }
+    }
+    private var events: [EventsForSportAndDate] = [] {
+        didSet {
+            tableView.reloadData()
+            updateEmptyState()
+        }
+    }
+    private var groupedEvents: [Tournament: [EventsForSportAndDate]] = [:]
     private let tableView = UITableView()
     private let leagueView = LeagueView()
-    private var exampleLeague: League?
+    private var exampleLeague: Tournament?
+    private let emptyStateLabel: UILabel = {
+        let label = UILabel()
+        label.text = "No events today"
+        label.textAlignment = .center
+        label.isHidden = true
+        return label
+    }()
     
     override func viewDidLoad() {
-        initializeData()
         addViews()
         setupConstraints()
         styleViews()
     }
     
-    private func initializeData() {
-        if let sportType = sportSlug?.asSportType {
-            exampleLeague = MockData.exampleLeague(forSport: sportType)
-            exampleEvents = MockData.exampleEvents(forSport: sportType)
-        } else { return }
+    private func updateEmptyState() {
+        emptyStateLabel.isHidden = !events.isEmpty
+        tableView.isHidden = events.isEmpty
+    }
+    
+    private func fetchEvents() {
+        guard let sportSlug = sportSlug, let date = date else { return }
+        let formattedDate = formatDate(date)
+        Task {
+            do {
+                let fetchedEvents = try await APIClient().fetchEventsAsync(sportSlug: sportSlug, date: formattedDate)
+                self.events = fetchedEvents
+                self.groupEventsByTournament(fetchedEvents)
+                self.tableView.reloadData()
```

Comment:
ne treba ti self

Reply:
i ovo tu napises samo i onda mozes pobrisat events gore
```suggestion
                groupedEvents = Dictionary(grouping: fetchedEvents, by: { $0.tournament })
                tableView.reloadData()
```

Reply:
+